tests/publiccloud/prepare_tools.pm

```diff
@@ -111,11 +111,11 @@ sub run {
     record_info('img-proof', $img_proof_ver);
     set_var('PUBLIC_CLOUD_IMG_PROOF_VER', $img_proof_ver =~ /img-proof, version ([\d\.]+)/);
 
-    my $terraform_version = '0.14.1';
+    my $terraform_version = '1.1.7';
```

Comment:
```suggestion
    my $terraform_version = 'get_required_var('PC_TOOLS_TERRAFORM_VERSION');
```

Out of the scope of this PR, but @asmorodskyi @jlausuch what do you think of passing the versions as variables instead of hardcoding them in the code?

Reply:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14490#pullrequestreview-910706525

Reply:
Honestly, relying on the code here to really KNOW which terraform is in the current publiccloud_tools image is not more reliable than checking the latest test run for the setting.
Having it as a variable however allows to create additional images with custom terraform versions and you can publish them as different images by also setting the `PUBLISH_HDD_1` variable.

Building a **custom** `publiccloud_tools` with any version other than the hardcoded once is currently not possible and I find that sad, especially because it was apparently requested here.

If the issue is to prevent that we are overwriting the existing pc tools image with a custom one, we can add easily a check that does not allow that from happening:

    die "Overwrite of publiccloud_tools image with custom terraform version is not allowed" if (get_var('PC_TOOLS_TERRAFORM_VERSION', undef) != undef) && (get_required_var('PUBLISH_HDD_1') ~= "publiccloud_tools");

or something similar.

Reply:
I would keep this PR as it was initially designed - upgrade of terraform version . @grisu48  if you have strong opinion about need of variable for terraform version let's create separate PR for this and continue discussion there